model/riscv_vmem_utils.sail

```diff
@@ -163,6 +163,49 @@ function vmem_write_addr(vaddr, width, data, acc, aq, rl, res) = {
   Ok(write_success)
 }
 
+/* Pointer Masking */
+
+/* Return the 2-bit PMM value based on the current effective privilege mode. */
+function get_pmm(eff_privilege : Privilege) -> bits(2) = match eff_privilege {
+  Machine    => mseccfg[PMM],
+  Supervisor => menvcfg[PMM],
+  User       => if currentlyEnabled(Ext_S) then senvcfg[PMM] else menvcfg[PMM],
+}
+
+/* Check whether PMM is applicable.
+ * NOTE: mstatus[MXR] == 1'b1 disables pointer masking in S and U mode, but not in M-mode.
+ */
+function is_pmm_applicable(typ : AccessType(ext_access_type), eff_privilege : Privilege) -> bool = typ != InstructionFetch() & (eff_privilege == Machine | mstatus[MXR] == 0b0)
```

Comment:
Minor style thing, but this is a pretty long line, normally we'd wrap it like this:

```
function is_pmm_applicable(typ : AccessType(ext_access_type), eff_privilege : Privilege) -> bool =
  typ != InstructionFetch() & (eff_privilege == Machine | mstatus[MXR] == 0b0)
```